internal/installation/app/app_darwin.go

```diff
@@ -0,0 +1,244 @@
+package app
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/assets"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/installation"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/osutils/user"
+	"github.com/ActiveState/cli/internal/strutils"
+)
+
+const (
+	execFileSource       = "exec.sh.tpl"
+	launchFileSource     = "com.activestate.platform.app.plist.tpl"
+	launchFileFormatName = "com.activestate.platform.%s.plist"
+	autostartFileSource  = "com.activestate.platform.autostart.plist.tpl"
+	iconFile             = "icon.icns"
+)
+
+var targetDirs = []string{
+	"/Contents",
+	"/Contents/MacOS",
+	"/Contents/Resources",
+}
+
+func (a *App) install() error {
+	// Create all of the necessary directories and files in a temporary directory
+	// Then move the temporary directory to the final location which for macOS will be the Applications directory
+	tmpDir, err := ioutil.TempDir("", "state-svc-")
+	if err != nil {
+		return errs.Wrap(err, "Could not create temporary directory")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	tmpAppPath := filepath.Join(tmpDir, fmt.Sprintf("%s.app", a.Name))
+	err = fileutils.Mkdir(tmpAppPath)
+	if err != nil {
+		return errs.Wrap(err, "Could not create .app directory")
+	}
+
+	for _, t := range targetDirs {
+		path := filepath.Join(tmpAppPath, t)
+		err = fileutils.Mkdir(path)
+		if err != nil {
+			return errs.Wrap(err, "Could not create directory at %s", path)
+		}
+	}
+
+	icon, err := assets.ReadFileBytes(a.options.IconFileSource)
+	if err != nil {
+		return errs.Wrap(err, "Could not read asset")
+	}
+
+	err = fileutils.WriteFile(filepath.Join(tmpAppPath, "Contents", "Resources", iconFile), icon)
+	if err != nil {
+		return errs.Wrap(err, "Could not write icon file")
+	}
+
+	err = a.createExecFile(filepath.Join(tmpAppPath, "Contents", "MacOS"))
+	if err != nil {
+		return errs.Wrap(err, "Could not create exec file")
+	}
+
+	err = a.createInfoFile(filepath.Join(tmpAppPath, "Contents"))
+	if err != nil {
+		return errs.Wrap(err, "Could not create info file")
+	}
+
+	installDir, err := installation.ApplicationInstallPath()
+	if err != nil {
+		return errs.Wrap(err, "Could not get installation path")
+	}
+
+	err = fileutils.MoveAllFiles(tmpDir, installDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not move .app to Applications directory")
+	}
```

Comment:
Can we try to use go/embed and put this whole file structure under assets? Then it should just be a matter of copying a single directory and doing some post-processing stuff.